CoreFoundation/String.subproj/CFCharacterSet.c

```diff
@@ -1417,15 +1418,14 @@ Boolean _CFCharacterSetInitWithCharactersInString(CFMutableCharacterSetRef cset,
             if ((*characters < 0xDC00UL) && (*(charactersLimit - 1) > 0xDBFFUL)) { // might have surrogate chars
                 while (characters < charactersLimit) {
                     if (CFStringIsSurrogateHighCharacter(*characters) || CFStringIsSurrogateLowCharacter(*characters)) {
-                        CFRelease(cset);
-                        cset = NULL;
+                        csetIsReady = false;
                         break;
                     }
                     ++characters;
                 }
             }
         }
-        if (NULL != cset) return cset;
```

Comment:
Somewhat surprised we were able to do this without getting a compiler warning or error of some kind.